### ISSUE TITLE
Node.js 8.0 Carbon End-Of-Life

### ISSUE THREAD

**Issue by a-ttila** [bigbluebutton/bigbluebutton#8937]
At my company we would like to use BigBlueButton, but one of our security guys told me that Node.js 8.0 Carbon has reached it's end of life and therefore it can be a security problem to use BigBlueButton.

Is there a plan to upgrade to Node.js 10.x LTS or 12.x LTS? And if there is, when?

Thank you for your amazing work!

**Comment by antobinary** [comment]
Hi @a-ttila 
On branch 'develop' we are working on what is to become BBB 2.3 and it is on node 12. We have not yet reached a 'beta' state yet. It is aiming Ubuntu 18.04 as well, so the update path is not as straight forward.

**Comment by agrisvv** [comment]
on prod server with 2.2.x  installed node 10.x- till now no problems.

**Comment by antobinary** [comment]
@agrisvv Just installing node10 on the machine does not affect bbb-html5.

We match the version of node that we use for Meteor (Meteor 1.8.2 -> node 8)
See `/usr/share/meteor/bundle/systemd_start.sh`

```
NODE_VERSION=node-v12.13.1-linux-x64
if [ "$DISTRO" == "xenial" ]; then
  NODE_VERSION=node-v8.15.1-linux-x64
fi
```
You will have to change NODE_VERSION and then restart bbb-html5

Also re-confirm by `journalctl -u bbb-html5.service` the first lines after bbb-html5 starts shows the node version

```
Mar 30 02:52:21 droplet-4689 systemd_start.sh[5114]: warn: SERVER STARTED.
Mar 30 02:52:21 droplet-4689 systemd_start.sh[5114]: ENV=production,
Mar 30 02:52:21 droplet-4689 systemd_start.sh[5114]: nodejs version=v8.15.1
```

I have not tried node 10 so I am curious what you encounter. We were planning on switching straight to (Meteor 1.9 -> node 12) but a few more things are in the way before sharing a beta with the community. Please post here your results with node 10!

**Comment by agrisvv** [comment]
yes i can confirm what it use /usr/share/8.x folder it is downloade by some package.
i was installed node 10 followed to  install instruction, just replace 8.x with 10.x but as i see it is not used and are just downloaded by other package.

**Comment by owlderup** [comment]
@antobinary Is there already an approximate release date for BBB 2.3 with node 12 support?

**Comment by antobinary** [comment]
Hi @owlderup looks like I missed your message

I am linking by a precise answer by Fred including steps others can take to help
https://github.com/bigbluebutton/bigbluebutton/issues/8474#issuecomment-784144216

**Comment by ffdixon** [comment]
Fixed in 2.3-dev.  See https://docs.bigbluebutton.org/dev/dev23.html
